Tests/scripts/collect_tests_and_content_packs.py

```diff
@@ -1048,14 +1048,14 @@ def is_test_uses_active_integration(integration_ids, conf=deepcopy(CONF)):
 
 def get_random_tests(tests_num, rand, conf=deepcopy(CONF), id_set=deepcopy(ID_SET), server_version='0'):
     """Gets runnable tests for the server version"""
-    tests = set([])
-    test_ids = conf.get_test_playbook_ids()
+    all_test_ids = conf.get_test_playbook_ids()
+    runnable_test_ids = [test_id for test_id in all_test_ids if is_test_runnable(test_id, id_set, conf, server_version)]
+    if len(runnable_test_ids) >= tests_num:
```

Comment:
if you dont have enough tests, grab what you have
```suggestion
    if len(runnable_test_ids) <= tests_num:
```